### DISCUSSION TITLE
FEATURE request: Configurable ipv6

### DISCUSSION THREAD

**Discussion by mews-se** [MichaIng/DietPi#7123] [Q&A] [answered]
This is neither a software nor image request, but a request to add an option in dietpi-config to actually configure ipv6 instead of just having a toggle for on/off.

The current function for ipv4 is excellent and if it could, it should be replicated for 6

Thanks!

**Comment by MichaIng** [comment, marked as answer]
Be aware that (GUA) IPv6 addresses naturally cannot be freely chosen, but they are bound to your public IPv6 prefix. There are 3 types:
- Every interface gets a local-link address (LLA) assigned, which is based on the MAC address, hence cannot be chosen.
- For Internet access, an interface must get a global unicast address (GUA), which as said, is based on the public IPv6 prefix, hence cannot be freely chosen either, but is assigned by the router via SLAAC.
- What could be freely chosen is a unique local address (ULA). However, these are disabled by default on most routers and generally not recommended, unless you require them for something particular. But while I know of use cases where a ULA is at least beneficial to have, I am not aware of any where it would be statically assigned by the client, but instead it would be also assigned by the router via SLAAC, with its local IPv6 prefix.

So the question is, which particular need to you have for a static IPv6 address, which can only be relevant for inter-LAN requests, where I would usually just use IPv4?

**Reply by mews-se** [reply]
Thanks for the answer, I must admit that my knowledge about ipv6 is less than yours :)

However I have some dietpi-instances running on locations with static ip:s aswell as ipv6 support - and I cant for the life of me figure out how to configure the correct way of making it work - hence my request :)

I think i messed around with SLAAC only to figure out that I manually needed to enter some stuff that was ISP specific - and it didnt work. 

So mostly its a PEBCAK issue but every step to ease the transition from 4 to 6 to the public is welcome :)

**Reply by MichaIng** [reply]
You mean you have DietPi instances which are directly attached to the Internet, without router? Not sure whether ISP <> device communication is the same as router <> device, but actually I would assume so, i.e. that the ISP is assigning the IPv6 address/prefix, with similar or the same automatic protocol than router would do.

What exactly did not work before you started changing IPv6 settings? Usually exactly no settings must be set or changed for IPv6, because really everything is done fully automatic. If IPv6 is enabled on kernel level, interfaces assign their LLA automatically, and then use SLAAC for everything else, if a router is there, without any config needed.

**Reply by mews-se** [reply]
There is a router, but even when configured with the "instructions" from the ISP - somehow it fails. Therefore I tried doing a direct connection without it, with the same results.

anyway, this is perhaps a bit overkill. If it works as intended - then all is well :)

**Reply by MichaIng** [reply]
How does it fail? And what do ISP instructions say about device configuration "behind" the router? As said, of course the router needs to be configured to get an IPv6 address and to forward the IPv6 prefix to devices via SLAAC (should be however the default). But the devices behind the router do not need any configuration, respectively at best must not have any configuration, to not mess with clean SLAAC.
